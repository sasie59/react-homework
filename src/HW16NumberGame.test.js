import React from 'react';
import { getByText, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW16NumberGame';


test('should fail number', () => {
  const { container, getByTestId } = render(<HW />);
  const input = container.querySelector('input');
  const fail = getByTestId('fail');
  userEvent.type(input,"r");
  expect(fail.textContent).toBe("答錯: 1");
  expect(container).toMatchSnapshot();
});

test('should bingo number', () => {
  const { container , getByTestId }  =render(<HW />);
  const input = container.querySelector('input');
  const random = getByTestId('random', 5);
  const bingo = getByTestId('bingo');
  userEvent.type(input, '5');
  expect(bingo.textContent).toBe("答對: 1");
  expect(container).toMatchSnapshot();
});




