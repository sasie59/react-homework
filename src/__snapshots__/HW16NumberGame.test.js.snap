// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should bingo number 1`] = `
<div>
  <div>
    <div
      data-testid="random"
    >
      隨機數字
      6
    </div>
    <input
      type="text"
      value=""
    />
    <div
      data-testid="bingo"
    >
      答對: 
      0
    </div>
    <div
      data-testid="fail"
    >
      答錯: 
      1
    </div>
  </div>
</div>
`;

exports[`should fail number 1`] = `
<div>
  <div>
    <div
      data-testid="random"
    >
      隨機數字
      1
    </div>
    <input
      type="text"
      value=""
    />
    <div
      data-testid="bingo"
    >
      答對: 
      0
    </div>
    <div
      data-testid="fail"
    >
      答錯: 
      1
    </div>
  </div>
</div>
`;
